sdk/core/azure-core-amqp/src/main/java/com/azure/core/amqp/implementation/handler/WebSocketsProxyConnectionHandler.java

```diff
@@ -0,0 +1,229 @@
+
+        logger.error(String.format("Failed to connect to url: '%s', proxy host: '%s'",
+            url.toString(), address.getHostString()), ioException);
+
+        if (proxySelector != null) {
+            proxySelector.connectFailed(url, address, ioException);
+        }
+    }
+
+    @Override
+    protected void addTransportLayers(final Event event, final TransportInternal transport) {
+        super.addTransportLayers(event, transport);
+
+        // Checking that the proxy configuration is not null and not equal to the system defaults option.
+        final ProxyImpl proxy = proxyConfiguration != null && !(proxyConfiguration == ProxyConfiguration.SYSTEM_DEFAULTS)
+            ? new ProxyImpl(getProtonConfiguration())
+            : new ProxyImpl();
+
+        // host name used to create proxy connect request
+        // after creating the socket to proxy
+        final String hostname = event.getConnection().getHostname();
+        final ProxyHandler proxyHandler = new ProxyHandlerImpl();
+        proxy.configure(hostname, null, proxyHandler, transport);
+
+        transport.addTransportLayer(proxy);
+
+        logger.info("addProxyHandshake: hostname[{}]", hostname);
+    }
+
+    private InetSocketAddress getProxyAddress() {
+        if (proxyConfiguration != null && proxyConfiguration.isProxyAddressConfigured()) {
+            return (InetSocketAddress) proxyConfiguration.getProxyAddress().address();
+        }
+
+        final URI serviceUri = createURI(amqpHostname, HTTPS_PORT);
+        final ProxySelector proxySelector = ProxySelector.getDefault();
+        if (proxySelector == null) {
+            throw logger.logExceptionAsError(new IllegalStateException(PROXY_SELECTOR_HAS_BEEN_MODIFIED));
+        }
+
+        final List<Proxy> proxies = proxySelector.select(serviceUri);
+        if (!isProxyAddressLegal(proxies)) {
+            throw logger.logExceptionAsError(new IllegalStateException(PROXY_SELECTOR_HAS_BEEN_MODIFIED));
+        }
+
+        final Proxy proxy = proxies.get(0);
+        return (InetSocketAddress) proxy.address();
+    }
+
+    private com.microsoft.azure.proton.transport.proxy.ProxyConfiguration getProtonConfiguration() {
+        final ProxyAuthenticationType type = getProtonAuthType(proxyConfiguration.getAuthentication());
+        final String username = proxyConfiguration.hasUserDefinedCredentials()
+            ? proxyConfiguration.getCredential().getUserName()
+            : null;
+        final String password = proxyConfiguration.hasUserDefinedCredentials()
+            ? new String(proxyConfiguration.getCredential().getPassword())
+            : null;
+
+        return new com.microsoft.azure.proton.transport.proxy.ProxyConfiguration(type,
+            proxyConfiguration.getProxyAddress(), username, password);
+    }
+
+    private ProxyAuthenticationType getProtonAuthType(com.azure.core.amqp.models.ProxyAuthenticationType type) {
```

Comment:
Would be better to import track 2 types and use fully qualified names for track 1 (or com.microsoft.azure) types.